fsw/src/generic_css_app.c

```diff
@@ -0,0 +1,614 @@
+        }
+        else
+        {
+            GENERIC_CSS_AppData.HkTelemetryPkt.DeviceErrorCount++;
+            CFE_EVS_SendEvent(GENERIC_CSS_REQ_HK_ERR_EID, CFE_EVS_ERROR, 
+                    "GENERIC_CSS: Request device HK reported error %d", status);
+        }
+    }
+    // Intentionally do not report errors if disabled
+
+    // Time stamp and publish housekeeping telemetry
+    CFE_SB_TimeStampMsg((CFE_SB_Msg_t *) &GENERIC_CSS_AppData.HkTelemetryPkt);
+    CFE_SB_SendMsg((CFE_SB_Msg_t *) &GENERIC_CSS_AppData.HkTelemetryPkt);
+    return;
+}
+*/
+
+/*
+** Collect and Report Device Telemetry
+*/
+void GENERIC_CSS_ReportDeviceTelemetry(void)
+{
+    int32 status = OS_SUCCESS;
+
+    /* Check that device is enabled */
+    if (GENERIC_CSS_AppData.HkTelemetryPkt.DeviceEnabled == GENERIC_CSS_DEVICE_ENABLED)
+    {
+        status = GENERIC_CSS_RequestData(GENERIC_CSS_AppData.Generic_cssI2c.handle, 
+            (GENERIC_CSS_Device_Data_tlm_t*) &GENERIC_CSS_AppData.DevicePkt.Generic_css,
+            (GENERIC_CSS_Device_HK_tlm_t*) &GENERIC_CSS_AppData.HkTelemetryPkt.DeviceHK);
+        if (status == OS_SUCCESS)
+        {
+            // QUESTION: Why does reporting device telemetry increment the number of devices?
+            GENERIC_CSS_AppData.HkTelemetryPkt.DeviceCount++;
+            /* Time stamp and publish data telemetry */
+            CFE_SB_TimeStampMsg((CFE_SB_Msg_t *) &GENERIC_CSS_AppData.DevicePkt);
+            CFE_SB_SendMsg((CFE_SB_Msg_t *) &GENERIC_CSS_AppData.DevicePkt);
+        }
+        else
+        {
+            GENERIC_CSS_AppData.HkTelemetryPkt.DeviceErrorCount++;
+            CFE_EVS_SendEvent(GENERIC_CSS_REQ_DATA_ERR_EID, CFE_EVS_ERROR, 
+                    "GENERIC_CSS: Request device data reported error %d", status);
+        }
+    }
+    /* Intentionally do not report errors if disabled */
+    return;
+}
+
+
+/*
+** Reset all global counter variables
+*/
+void GENERIC_CSS_ResetCounters(void)
+{
+    GENERIC_CSS_AppData.HkTelemetryPkt.CommandErrorCount = 0;
+    GENERIC_CSS_AppData.HkTelemetryPkt.CommandCount = 0;
+    GENERIC_CSS_AppData.HkTelemetryPkt.DeviceErrorCount = 0;
+    GENERIC_CSS_AppData.HkTelemetryPkt.DeviceCount = 0;
+    return;
+} 
+
+
+/*
+** Enable Component
+** TODO: Edit for your specific component implementation
+*/
+void GENERIC_CSS_Enable(void)
+{
+    int32 status = OS_SUCCESS;
+
+    /* Check that device is disabled */
+    if (GENERIC_CSS_AppData.HkTelemetryPkt.DeviceEnabled == GENERIC_CSS_DEVICE_DISABLED)
+    {
+        /*
+        ** Initialize hardware interface data
+        ** TODO: Make specific to your application depending on protocol in use
+        ** Note that other components provide examples for the different protocols available
+        */ 
+        GENERIC_CSS_AppData.Generic_cssI2c.handle = GENERIC_CSS_CFG_HANDLE;
+        GENERIC_CSS_AppData.Generic_cssI2c.isOpen = PORT_CLOSED; // Question: should this be open?
```

Comment:
Should only be editing these values during the app_init phase actually, but in that yes we should just initialize the first time to closed.  Also remove TODO above.